test/lit/passes/Oz.wast

```diff
@@ -150,4 +158,96 @@
       (i32.const 0x80000000)
     )
   )
+
+  ;; CHECK:      (func $11 (; has Stack IR ;) (param $0 i32) (result i32)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.get $0)
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $11 (export "eliminate-redundant-checks-1") (param $0 i32) (result i32)
+    (if
+      (local.get $0)
+      (if
+        (local.get $0)
+        (return (local.get $0))
+      )
+    )
+    (i32.const 0)
+  )
+
+  ;; CHECK:      (func $12 (; has Stack IR ;) (param $0 i32) (param $1 i32) (result i32)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (local.tee $1
+  ;; CHECK-NEXT:    (local.get $0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $1)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $12 (export "eliminate-redundant-checks-2") (param $0 i32) (param $1 i32) (result i32)
+    (if
+      (local.tee $1 (local.get $0))
+      (if
+        (local.get $1)
+        (return (local.get $1))
+      )
+    )
+    (i32.const 0)
+  )
+
+  ;; CHECK:      (func $13 (; has Stack IR ;) (param $0 i32) (param $1 i32) (result i32)
+  ;; CHECK-NEXT:  (if
+  ;; CHECK-NEXT:   (select
+  ;; CHECK-NEXT:    (local.get $1)
+  ;; CHECK-NEXT:    (i32.const 0)
+  ;; CHECK-NEXT:    (local.get $0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:   (return
+  ;; CHECK-NEXT:    (local.get $0)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (i32.const 0)
+  ;; CHECK-NEXT: )
+  (func $13 (export "eliminate-redundant-checks-skip-1") (param $0 i32) (param $1 i32) (result i32)
```

Comment:
In this file the tests that are optimized use an `if` while the skipped ones use a `select`. Please add one that is optimized with a `select`, so we actually directly test a `select` of a `tee` (and not indirectly by turning an `if` into a `select` first).

Reply:
Done!

Reply:
Tests look good but I think you need to run the auto updater script too to get the CHECKs

Reply:
I ran an auto update script. Simply "$11" / "$11a" and "$12" / "$12a"  are optimized in the same function's body and then deduplicated

Reply:
Ok, I see, right...